sound/soc/sof/intel/hda-bus.c

```diff
@@ -106,3 +126,12 @@ void sof_hda_bus_init(struct hdac_bus *bus, struct device *dev,
 #endif
 
 }
+
+/*
+ * This can be used for both with/without hda link support.
+ */
+void sof_hda_bus_deinit(struct hdac_bus *bus)
+{
+	if (bus)
```

Comment:
this test is useless, in the caller the code uses iounmap(bus->remap_addr); so if the bus was NULL then you'd get a dereference earlier.